src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+﻿// THESE ARE THROWAWAY STUBS backing the mock-up example.
+// No need to review this file
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Sarif.WorkItems.Sample
+{
+    public partial class Program
+    {
+
+#if PREVIOUS
+        private static void NonFunctioningEndToEndApiUsageExample()
+        {
+            // 1. Retrieve the current SARIF and its baseline equivalent
+            SarifLog baselineSarif = RetrieveBaselineSarif();
+            SarifLog currentSarif = RetrieveCurrentSarif();
+
+            // 2. Match the SARIF files
+            ISarifLogBaseliner resultMatcher = SarifLogBaselinerFactory.CreateSarifLogBaseliner(SarifBaselineType.Standard);
+
+            SarifLog matchedLog = new SarifLog
+            {
+                Runs = new[]
+                {
+                    resultMatcher.CreateBaselinedRun(baselineSarif.Runs[0], currentSarif.Runs[0])
+                }
+            };
+
+            // 3. Make sure everything has a GUID and that we have a map back to each result from it
+            var guidToMatchedResultMap = new Dictionary<string, Result>();
+            foreach (Result result in matchedLog.Runs[0].Results)
+            {
+                if (result.Guid == null)
+                {
+                    result.Guid = Guid.NewGuid().ToString();
+                }
+                guidToMatchedResultMap[result.Guid] = result;
+            }
+
+            var adoClient = new AzureDevOpsClient();
+            var filingTarget = new AzureDevOpsFilingTarget(adoClient);
+
+            var workItemFiler = new WorkItemFiler(
+                filingTarget,
+                filteringStrategy: new NewResultsFilteringStrategy(),
+                groupingStrategy: new OneResultPerWorkItemGroupingStrategy());
+
+            string logFilePath = Path.GetTempFileName();
+            File.WriteAllText(logFilePath, JsonConvert.SerializeObject(matchedLog));
+
+            // IMPORTANT! There's no facility defined yet for creating/configuring titles, area paths, etc.
+
+            try
+            {
+                IEnumerable<ResultGroup> resultGroups = workItemFiler.FileWorkItems(logFilePath).Result;
+                foreach (ResultGroup resultGroup in resultGroups)
+                {
+                    foreach (Result result in resultGroup.Results)
+                    {
+                        bool matched = guidToMatchedResultMap.TryGetValue(result.Guid, out Result originalResult);
+                        Debug.Assert(matched);
+                        originalResult.WorkItemUris = result.WorkItemUris;
```

Comment:
>originalResult [](start = 24, length = 14)

Ok, reference equals one of the `Result` objects in the `matchedLog`. Worth a comment.